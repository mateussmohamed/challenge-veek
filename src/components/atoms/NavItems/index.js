import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const StyledNavItems = styled.div`
  @media only screen and (min-width: 0em) {
    a,
    button {
      display: none;
    }

    .logo {
      display: block;
      text-align: center;
    }
  }
  @media only screen and (min-width: 64em) {
    width: 630px;
    position: relative;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-evenly;

    a,
    button {
      display: block;
    }
  }
`

const StyledReponsiveItems = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  width: 100%;
  max-width: 260px;
  height: 100vh;
  background-color: ${props => props.theme.colors.primary};
  position: fixed;
  top: 0;
  left: 0;
  z-index: 3;
  padding: 50px 0;
  transition: all 0.5s ease 0s;
  transform: translate3d(-100%, 0px, 0px);

  &.active {
    transform: translate3d(0px, 0px, 0px);
  }

  a {
    color: #fff;
  }

  @media only screen and (min-width: 64em) {
    display: none;
  }
`

const NavItems = props => (
  <Fragment>
    <StyledNavItems {...props}>{props.children}</StyledNavItems>
    <StyledReponsiveItems {...props}>{props.children}</StyledReponsiveItems>
  </Fragment>
)

NavItems.propTypes = {
  children: PropTypes.node.isRequired,
}

export default NavItems
