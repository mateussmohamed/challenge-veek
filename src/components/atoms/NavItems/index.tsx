import React, { Fragment } from 'react'
import styled from 'styled-components'

const StyledNavItems = styled.div`
  @media only screen and (min-width: 0em) {
    a,
    button {
      display: none;
    }

    .logo {
      display: block;
      text-align: center;
    }
  }
  @media only screen and (min-width: 64em) {
    width: 630px;
    position: relative;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-evenly;

    a,
    button {
      display: block;
    }
  }
`

interface StyledReponsiveItemsProps {
  active?: boolean
}

const StyledReponsiveItems = styled.div<StyledReponsiveItemsProps>`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  width: 100%;
  max-width: 260px;
  height: 100vh;
  background-color: ${props => props.theme.colors.primary};
  position: fixed;
  top: 0;
  left: 0;
  z-index: 3;
  padding: 50px 0;
  transition: all 0.5s ease 0s;
  transform: translate3d(-100%, 0px, 0px);

  ${props => props.active && 'transform: translate3d(0px, 0px, 0px);'}

  a {
    color: #fff;
  }

  @media only screen and (min-width: 64em) {
    display: none;
  }
`

interface NavItemsProps {
  children: React.ReactNode
  active?: boolean
}

export const NavItems = (props: NavItemsProps) => (
  <Fragment>
    <StyledNavItems {...props}>{props.children}</StyledNavItems>
    <StyledReponsiveItems {...props}>{props.children}</StyledReponsiveItems>
  </Fragment>
)
