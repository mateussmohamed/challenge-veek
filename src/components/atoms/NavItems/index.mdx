---
name: NavItems
menu: Atoms
---

import { Playground, PropsTable } from 'docz'
import NavItems from './index'

# NavItems
<PropsTable of={NavItems} />

<Playground>
  <NavItems />
</Playground>
