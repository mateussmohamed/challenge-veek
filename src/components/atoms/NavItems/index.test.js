import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import NavItems from './index'

describe('[<NavItems />]', () => {
  const wrap = (props = {}) => shallow(<NavItems {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap()

    expect(wrapper).toMatchSnapshot()
  })
})
