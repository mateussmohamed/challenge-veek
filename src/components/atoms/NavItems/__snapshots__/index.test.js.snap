// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<NavItems />] matches to snapshot 1`] = `
<NavItems__StyledNavItems
  color="#fff"
>
  I am a NavItems
</NavItems__StyledNavItems>
`;

exports[`[<NavItems />] when pass props when pass color prop 1`] = `
<NavItems__StyledNavItems
  color="#283583"
/>
`;
